recipes/txaio/meta.yaml

```diff
@@ -0,0 +1,42 @@
+{% set name = "txaio" %}
+{% set version = "2.5.2" %}
+{% set sha256 = "321d441b336447b72dbe81a4d73470414454baf0543ec701fcfecbf4dcbda0fe" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - six
```

Comment:
So there isn't really a way to handle extra requirements in `conda` ATM. I could see a few options.

1. Don't provide them and let users install them.
2. Provide some that you deem crucial.
3. Provide all of them.
4. Add other packages for each of the extra installs that require this one and pull in the extras.

Probably 3 is too extreme. 1 is technically fine, but maybe 2 is better. Really making the decision about how many to require here is under your discretion as a maintainer. 4 might be a viable solution, but seems to add more maintenance as opposed to solving them here somehow.

Some time ago, I raised issue ( https://github.com/conda/conda/issues/3299 ) to address optional dependencies that require pinnings. This kind of fits given the version constraints. Though it wouldn't necessarily solve this problem completely.